### PR TITLE
(DOCSP-8764): Port auth provider: Facebook

### PR DESCRIPTION
**Pull request by cbush**
- Example SDK snippets cut... for now
- Step files to be replaced

## JIRA
https://jira.mongodb.org/browse/DOCSP-8764

## Staged
https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker/port-fb/authentication/facebook

### CODE REVIEW

File: source/includes/deployment-region-auth-callback-urls.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+.. cssclass:: config-table
+.. list-table::
+   :header-rows: 1
+   :widths: 1 4
+
+   * - Region
+     - Stitch Authentication Callback URL
```
Comment (nlarew) [inline source/includes/deployment-region-auth-callback-urls.rst]: Should be Realm here. Not sure what the actual URLs here will be but I've noted that we should follow up on it.

File: conf.py [hunk 2]
```diff
@@ -64,6 +64,7 @@
     # MongoDB Docs Sites
     'manual': ('http://docs.mongodb.org/manual%s', ''),
     'atlas': ('https://docs.atlas.mongodb.com%s',''),
+    'fb-dev-docs': ('https://developers.facebook.com/docs/%s', ''),
```
Comment (nathan-contino-mongo) [inline conf.py]: IIRC andrew mentioned that all of the link directives are now defined in the snooty [rstspec](https://github.com/mongodb/snooty-parser/blob/f651721f62259791ecae21f262956f4e41ab6f87/snooty/rstspec.toml) Or is this something else (could very easily be something I'm not yet familiar with)

Comment (i80and) [reply conf.py]: Correct: conf.py is no longer used.

We would like to support adding extlinks in snooty.toml when scheduling permits, but for now, please file a bug with the name and target URL.

Comment (cbush) [reply conf.py]: I will still leave it for now since I've been using oldgen to share the site with the stakeholders (until TOC is implemented)

Comment (nathan-contino-mongo) [reply conf.py]: Interestingly enough @cbush it looks like fb-dev-docs is indeed in that file. See the line beneath my additions in this PR: https://github.com/mongodb/snooty-parser/pull/98

Comment (nathan-contino-mongo) [reply conf.py]: On that note I will also use oldgen to preview the site and maintain conf.py

File: source/includes/steps-auth-facebook-app-setup.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ref: auth-create-facebook-app
+content: |
+   Follow Facebook's `official guide
+   <https://developers.facebook.com/quickstarts/>`_ to create a new
+   Facebook app.
+---
+title: Enable Facebook Login
+ref: auth-enable-facebook-login
+content: |
+  From the app's :guilabel:`Dasboard` view, find the :guilabel:`Facebook
```
Comment (nathan-contino-mongo) [inline source/includes/steps-auth-facebook-app-setup.yaml]:
```suggestion
  From the app's :guilabel:`Dashboard` view, find the :guilabel:`Facebook
```

File: source/includes/deployment-region-auth-callback-urls.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
+.. cssclass:: config-table
+.. list-table::
+   :header-rows: 1
+   :widths: 1 4
+
+   * - Region
+     - Realm Authentication Callback URL
+
+   * - | **Global**
+     - .. code-block:: none
```
Comment (nathan-contino-mongo) [inline source/includes/deployment-region-auth-callback-urls.rst]: the black and red rendering for this seems a little off. Can we change the code-block to something better for urls? (I seem to remember a URL target of some kind)

File: source/authentication/facebook.txt [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+and configuring the provider to connect with the app.
+
+.. include:: /includes/steps/auth-facebook-app-setup.rst
+
+.. _auth-facebook-client-sdks:
+
+Authenticate a User
+~~~~~~~~~~~~~~~~~~~
+
+TBD.
```
Comment (nathan-contino-mongo) [inline source/authentication/facebook.txt]: Should we settle on a consistent label for this so it's easy to search for all of the locations where we're missing examples? I'm fine with "TBD" or "TODO" or "MISSING EXAMPLE HERE" or anything that we can easily grep.

File: source/includes/deployment-region-auth-callback-urls.rst [hunk 6]
```diff
@@ -0,0 +1,36 @@
+.. cssclass:: config-table
+.. list-table::
+   :header-rows: 1
+   :widths: 1 4
+
+   * - Region
+     - Realm Authentication Callback URL
+
+   * - | **Global**
```
[the remark this answers is not in the document]
Comment (cbush) [reply source/includes/deployment-region-auth-callback-urls.rst]: Noticed that. I brought it up in snooty-gtm.

File: source/includes/deployment-region-auth-callback-urls.rst (hunk 4 above)
Comment (cbush) [reply source/includes/deployment-region-auth-callback-urls.rst]: I presume they used code-block to allow it to be copyable? I will try `text` to eliminate the syntax highlighting.

**Pull request merged**